### PR TITLE
Java: Move SensitiveLoggerConfig source to extensible format

### PR DESCRIPTION
**Pull request by chmodxxx**
We have a need to do customizations on the sensitivelogger query where we need to extend the source of SensitiveLoggerConfiguration. Putting up this PR to see if I got things correctly .

### CODE REVIEW

File: java/ql/lib/semmle/code/java/security/SensitiveLoggingQuery.qll [hunk 1]
```diff
@@ -26,6 +29,12 @@ class CredentialExpr extends VarAccess {
   }
 }
 
+class CredentialExprSource extends SensitiveLoggerSource {
+  CredentialExprSource(){
+    this.asExpr() instanceof CredentialExpr
+  }
```
Comment (aschackmull) [inline java/ql/lib/semmle/code/java/security/SensitiveLoggingQuery.qll:35]: Formatting fix:
```suggestion
  CredentialExprSource() { this.asExpr() instanceof CredentialExpr }
```

File: java/ql/lib/semmle/code/java/security/SensitiveLoggingQuery.qll [hunk 2]
```diff
@@ -59,7 +66,7 @@ deprecated class SensitiveLoggerConfiguration extends TaintTracking::Configurati
 
 /** A data-flow configuration for identifying potentially-sensitive data flowing to a log output. */
 module SensitiveLoggerConfig implements DataFlow::ConfigSig {
-  predicate isSource(DataFlow::Node source) { source.asExpr() instanceof CredentialExpr }
+  predicate isSource(DataFlow::Node source) { sourc instanceof SensitiveLoggerSource }
```
Comment (aschackmull) [inline java/ql/lib/semmle/code/java/security/SensitiveLoggingQuery.qll:69]:
```suggestion
  predicate isSource(DataFlow::Node source) { source instanceof SensitiveLoggerSource }
```

**Review by aschackmull — APPROVED** [review]
LGTM!

File: java/ql/lib/semmle/code/java/security/SensitiveLoggingQuery.qll [hunk 3]
```diff
@@ -26,6 +29,10 @@ class CredentialExpr extends VarAccess {
   }
 }
 
+class CredentialExprSource extends SensitiveLoggerSource {
```
Comment (aschackmull) [inline java/ql/lib/semmle/code/java/security/SensitiveLoggingQuery.qll:32]: No need for this to be public.
```suggestion
private class CredentialExprSource extends SensitiveLoggerSource {
```

**Comment by owen-mc** [comment]
Do you think this needs a change note, @aschackmull ?

**Comment by aschackmull** [comment]
> Do you think this needs a change note, @aschackmull ?

Yeah, I was just thinking about that as well - it might indeed be best to include that.

**Comment by owen-mc** [comment]
@chmodxxx There is guidance on how to write a change note [here](https://github.com/github/codeql/blob/main/docs/change-notes.md). You can also look at previous examples. Please put it in the same folder as [this one](https://github.com/github/codeql/blob/main/java/ql/src/change-notes/2024-07-07-android-application-heuristic-updated.md).

**Review by owen-mc — APPROVED** [review]

**Comment by chmodxxx** [comment]
Hmm, not sure what's wrong with the CI integration test for Windows. Is it a flake ?

**Comment by owen-mc** [comment]
It doesn't seem like it could be related to your PR. I've restarted it.

**Pull request merged**